### ISSUE TITLE
Export sanitizeRestProps in index.js files

### ISSUE THREAD

**Issue by frisi** [marmelab/react-admin#4959]
for implementing custom fields, inputs, form components we often have to get rid of some properties before passing props to child-components

react-admin defines some helper methods for this:
eg when writing a field and use typography inside it (as it is done for numberinput here
https://github.com/marmelab/react-admin/blob/v3.6.1/packages/ra-ui-materialui/src/field/NumberField.tsx#L72)

to have this utility functions in place in my typescript project i'd suggest to export them in https://github.com/marmelab/react-admin/blob/v3.6.1/packages/ra-ui-materialui/src/field/sanitizeRestProps.ts#L3

eg like this:

```
import sanitizeFieldRestProps from './sanitizeRestProps';

export {
    sanitizeFieldRestProps,
```

iiuc w/o this export i currently have to import from the `esm` folder in my typescript project
```
import sanitizeFieldRestProps from 'ra-ui-materialui/esm/field/sanitizeRestProps'
```

(same applies to sanitizeRestProps for Inputs)

**Comment by djhi** [comment]
We are most probably going to simplify that a lot. See #4952 for example.

**Comment by fzaninotto** [comment]
#4952 will not remove prop injection to keep backward compatibility, so this enhancement will be interesting until we have a 4.0.

**Comment by fzaninotto** [comment]
I think naming and documenting all the sanitize rest props functions would be counter productive, as there are almost one per component.

**Comment by fzaninotto** [comment]
sorry, disregard my previous comment. Reopening.

**Comment by jdemangeon** [comment]
Then, what should we do @fzaninotto ? This issue has been open more than 2 months ago

**Comment by fzaninotto** [comment]
Yes, we should export and document the `sanitizeFieldRestProps` method.

**Comment by tanpld** [comment]
> Yes, we should export and document the `sanitizeFieldRestProps` method.

Just encountered this problem. Thank you for your consideration

**Comment by fzaninotto** [comment]
Fixed by #5392
